### ISSUE TITLE
Contract Interfaces don't have .balance member

### ISSUE THREAD

**Issue by fubuloubu** [vyperlang/vyper#2061]
### What's your issue about?
Contract Interfaces are an extension to an `address`, so it should have most (if not all) of the `address` members:
* `address.balance`
* `address.codehash`
* `address.codesize`
* `address.is_contract`

Error:
```python
interface PiggyBank:
    def deposit(): payable

piggy: PiggyBank

@public
def foo() -> uint256:
    self.piggy.deposit()
    return self.piggy.balance  
```
```
E       vyper.exceptions.UnknownAttribute: PiggyBank has no member 'balance'                                                             
E       function 'foo', line 10:11                                                                                                       
E             9     self.piggy.deposit()                                                                                                 
E       ---> 10     return self.piggy.balance                                                                                            
E       -------------------^                                                                                                             
E            11       
```

### How can it be fixed?
I believe `assert address.is_contract` is inserted prior to any external call to the interface, so clearly it is useful. I think it's best that this automatic check continues to exist only on call (and not on assignment, as this precludes counterfactual instantiations), so no change required there, but someone might want to add additional checks after assignment, for example:
```python
piggy: PiggyBank = PiggyBank(piggy_address)
assert piggy.is_contract
# OR
assert piggy.codehash == self.expected_piggy_codehash
# OR
assert piggy.codesize == 42  # Well, maybe not this one as much
```
I know it's just as easy to perform these checks prior to assignment (and probably more advisible to do so), but it just seems inconsistent not to have access to the `address` API for interfaces

**Comment by iamdefinitelyahuman** [comment]
The issue I see is that `balance` isn't a reserved keyword in solidity, so some contracts will contain a `balance` method.  For example, [iEarn](https://github.com/iearn-finance/itoken/blob/master/contracts/YDAI.sol#L506).

What if we insated add an `address` member, that gives access to the object's address type?

```python
piggy: PiggyBank = PiggyBank(piggy_address)
assert piggy.address.is_contract
assert piggy.address.codehash == self.expected_piggy_codehash
```

**Comment by fubuloubu** [comment]
not a bad idea!

**Comment by charles-cooper** [comment]
this was implemented in 6c8078ade (https://github.com/vyperlang/vyper/pull/2029)
